lib/APMSpan.js

```diff
@@ -65,7 +73,43 @@ class APMSpan {
             nodeType = 'Consumer';
         }
 
-        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId);
+        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId ? [{
+            value: consumerCorrelationId,
+            scope: 'Interaction',
+        }] : undefined);
+    }
+
+    _initFollowsFromSpanContext(parentReferences) {
+        function findRoot(span) {
+            let rootSpan = span;
+            while (rootSpan.getParentId()) {
+                const node = span.getTrace().findNode(rootSpan.getParentId());
+                if (node) {
+                    rootSpan = node.getSpan();
+                } else {
+                    break;
+                }
+            }
+
+            return rootSpan;
+        }
+
+        const id = utils.generateSpanId();
+        let traceId = null;
+
+        const correlationIds = [];
+        for (let i = 0; i < parentReferences.length; i += 1) {
+            const parentReference = parentReferences[i] instanceof APMSpan ? parentReferences[i].context() : parentReferences[i];
```

Comment:
Ok the logic should be:
1) If no 'childOf' refrerence then create a Trace
2) If one or more childOf references, then span/node should be added as child node of the first referenced span (ignore other childOf refs for now)
3) If one or more followsFrom refs are defined, then each should be converted to a CausedBy correlation id on the new node.